CRM/Mailing/BAO/MailingJob.php

```diff
@@ -478,6 +478,40 @@ public function queue($testParams = NULL) {
   }
 
   /**
+   * Build a join and where part for a query
+   *
+   * @param $contact_id
+   * @return array - the first key is join part of the query and the second key is the where part of the query
+   */
+  public function buildAcl($contact_id) {
```

Comment:
Yes it is duplicate code. However I am not comfortable where to put the function. To me it does not make sense to put it either in CRM_Mailing_BAO_Mailing or CRM_Mailing_BAO_MailingJob and then reference it to one or the other. That sound a bit like coupling the code rather than loosening.